components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/WorkflowsApiServiceImpl.java

```diff
@@ -0,0 +1,207 @@
+     * @return
+     */
+    @Override
+    public Response workflowsExternalWorkflowRefGet(String externalWorkflowRef, MessageContext messageContext)
+            throws APIManagementException {
+        WorkflowInfoDTO workflowinfoDTO;
+        try {
+            Workflow workflow;
+            String status = "CREATED";
+            String tenantDomain = RestApiCommonUtil.getLoggedInUserTenantDomain();
+            APIAdmin apiAdmin = new APIAdminImpl();
+            workflow = apiAdmin.getworkflowReferenceByExternalWorkflowReferenceID(externalWorkflowRef, status, tenantDomain);
+            workflowinfoDTO = WorkflowMappingUtil.fromWorkflowsToInfoDTO(workflow);
+            return Response.ok().entity(workflowinfoDTO).build();
+        } catch (APIManagementException e) {
+            RestApiUtil.handleInternalServerError("Error while retrieving workflow request by the " +
+                    "external workflow reference. ", e, log);
+        }
+        return null;
+    }
+
+     /*
+     * @param limit        maximum number of workflow returns
+     * @param offset       starting index
+     * @param accept       accept header value
+     * @param workflowType is the the type of the workflow request. (e.g: Application Creation, Application Subscription etc.)
+     * @return
+     */
+    @Override
+    public Response workflowsGet(Integer limit, Integer offset, String accept, String workflowType,
+                                 MessageContext messageContext) throws APIManagementException {
+        limit = limit != null ? limit : RestApiConstants.PAGINATION_LIMIT_DEFAULT;
+        offset = offset != null ? offset : RestApiConstants.PAGINATION_OFFSET_DEFAULT;
+        String tenantDomain = RestApiCommonUtil.getLoggedInUserTenantDomain();
+        String query = "";
+        Map<String, Object> result;
+        APIProvider apiProvider = RestApiCommonUtil.getLoggedInUserProvider();
+        boolean isMore = true;
+        int start = 0;
+        List<APIIdentifier> apiIdentifiers = null;
+        while(isMore) {
+            result = apiProvider.searchPaginatedAPIs(query, tenantDomain, 0, start+100, RestApiConstants.DEFAULT_SORT_CRITERION, RestApiConstants.DESCENDING_SORT_ORDER );
+            apiIdentifiers  =  new ArrayList<>();
+            Set<API> apis = (Set<API>) result.get("apis");
+            for(API api : apis){
+                apiIdentifiers.add(api.getId());
+            }
+            isMore = Boolean.parseBoolean(result.get("isMore").toString());
+            if(isMore) {
+                start = start + 100;
+            }
+        }
+        WorkflowListDTO workflowListDTO;
+        try {
+            Workflow[] workflows;
+            List<Workflow> workflowList;
+            String status = "CREATED";
+            APIAdmin apiAdmin = new APIAdminImpl();
+            if(workflowType != null) {
+                if (workflowType.equals("SUBSCRIPTION_CREATION")) {
```

Comment:
Move these to constants

Reply:
These string are used only one time in the class.